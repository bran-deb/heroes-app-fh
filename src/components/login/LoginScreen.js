import React from 'react'

export const LoginScreen = () => {
    return (
        <div>
            <h1>LoginScreen</h1>
        </div>
    )
}
