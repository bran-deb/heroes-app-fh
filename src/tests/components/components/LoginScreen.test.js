import { mount } from 'enzyme'
import { MemoryRouter, Routes, Route } from 'react-router-dom'
import { AuthContext } from '../../../auth/AuthContext'
import { LoginScreen } from '../../../components/login/LoginScreen'
import { types } from '../../../types/types'

const mockNavigate = jest.fn()

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate
}))


describe('Pruebas en LoginScreen', () => {
    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false,
        }
    }
    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter initialEntries={['/login']}>
                {/* <LoginScreen /> */}
                <Routes>
                    <Route path="/login" element={<LoginScreen />} />
                </Routes>
            </MemoryRouter>
        </AuthContext.Provider>
    )

    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('debe de realizar el dispatch y la navegacion', () => {
        const handleClick = wrapper.find('button').prop('onClick')
        handleClick()

        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.login,
            payload: {
                name: 'brand'
            }
        })
        expect(mockNavigate).toHaveBeenCalledWith('/', { replace: true })

        localStorage.setItem('lastPath', '/dc')
        handleClick()

        expect(mockNavigate).toHaveBeenCalledWith('/dc', { replace: true })
    })

    test("Debe realizar el dipatch y la navegaciÃ³n", () => {
        // wrapper.find('button').prop('onClick')()
        const handleClick = wrapper.find('button').prop('onClick')
        handleClick()
        expect(contextValue.dispatch).toHaveBeenCalledWith({
            'type': types.login,
            payload: {
                name: 'brand'
            }
        })
        expect(mockNavigate).toHaveBeenCalledWith('/', { replace: true })
        localStorage.setItem('lastPath', '/dc')
        handleClick();
        expect(mockNavigate).toHaveBeenCalledWith("/", { replace: true })
    })
})
