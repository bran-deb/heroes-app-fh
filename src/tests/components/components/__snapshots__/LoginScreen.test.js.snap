// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en LoginScreen debe de mostrarse correctamente 1`] = `
<div
  className="container mt-5 animate__animated animate__fadeIn"
>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    login
  </button>
</div>
`;
